app/controllers/admin/categories_controller.rb

```diff
@@ -10,7 +18,7 @@ def create
       else
         flash[:danger] = t ".message_danger"
       end
-      redirect_to categories_url
+      redirect_to request.referrer
```

Comment:
Can you try using redirect_back fallback_location method
[https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back](https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back)